File: Nikola.py
import sys
best = 10000000000000

def solve(a, last_jump, cost, pos):
    global best
    if pos < 0 or pos >= len(a):
        return  10000000000000 
    if pos == len(a)-1:
        ans = cost + a[pos] 
        if ans <  best:
            best = ans
        return ans 
    if cost > best:
        return 10000

    return min(solve(a, last_jump, cost+a[pos], pos-last_jump), solve(a, last_jump+1, cost+a[pos], pos + last_jump + 1)) 

n = int(sys.stdin.readline())
a = []
for i in sys.stdin:
    a.append(int(i))
print(solve(a, 1, 0, 1)) 

File: Nikola.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
#define endl "\n"
#define rep(i,a,b) for(int i = a; i<b; i++)

int best = 1000000000; 

inline int solve(vector<int> a, int last_jump, int cost, int pos){
    if(pos < 0 || pos >= a.size())
        return  1000000000; 

    if(pos == a.size()-1){
        int ans = cost + a[pos]; 
        if(ans < best)
            best = ans; 
        return ans; 
    }

    return min(solve(a, last_jump, cost+a[pos], pos-last_jump), solve(a, last_jump+1, cost+a[pos], pos + last_jump + 1)); 
}

    int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    int n, i, j;
    cin >> n; 

    vector<int> v; 

    rep(i, 0, n){
       cin >> j; 
       v.push_back(j);  
    }
        

    cout << solve(v, 1, 0, 1) << endl; 
    return 0;
}   